src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/Logging/RedactedLogValueIntegrationTest.cs

```diff
@@ -25,6 +28,83 @@ private static class EventIds
             public static readonly EventId RequestPipelineResponseHeader = new EventId(103, "RequestPipelineResponseHeader");
         }
 
+                [Fact]
+        public async Task RedactLoggedHeadersNotCalled_AllValuesAreRedactedBeforeLogging()
+        {
+            // Arrange
+            var sink = new TestSink();
+
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddLogging();
+            serviceCollection.AddSingleton<ILoggerFactory>(new TestLoggerFactory(sink, enabled: true));
+
+            // Act
+            serviceCollection
+                .AddHttpClient("test")
+                .ConfigurePrimaryHttpMessageHandler(() => new TestMessageHandler());
+
+            // Assert
+            var services = serviceCollection.BuildServiceProvider();
+
+            var client = services.GetRequiredService<IHttpClientFactory>().CreateClient("test");
+
+            var request = new HttpRequestMessage(HttpMethod.Get, "http://example.com");
+            request.Headers.Authorization = new AuthenticationHeaderValue("fake", "secret value");
+            request.Headers.CacheControl = new CacheControlHeaderValue() { NoCache = true, };
+
+            await client.SendAsync(request);
+
+            var messages = sink.Writes.ToArray();
+
+            var message = Assert.Single(messages.Where(m =>
+            {
+                return
+                    m.EventId == EventIds.RequestPipelineRequestHeader &&
+                    m.LoggerName == OuterLoggerName;
+            }));
+            Assert.StartsWith(LineEndingsHelper.Normalize(
+@"Request Headers:
+Authorization: *
+Cache-Control: *
+"), message.Message);
```

Comment:
```suggestion
            Assert.StartsWith(LineEndingsHelper.Normalize(
                """
                Request Headers:
                Authorization: *
                Cache-Control: *
                """),
                message.Message);
```

For future reference, these could use raw string literals